### ISSUE TITLE
Split off cluster management into a gem

### ISSUE THREAD

**Issue by betamatt** [infochimps-labs/ironfan#2]
It would be very convenient to have the cluster management plugin part of the project split off into its own gem.  This would include the knife plugin and cluster DSL but not the cookbooks and hadoop-specific kludges.  Right now, using this project to manage a non-hadoop cluster is much more difficult than it could be- 1. install gem 2. require in knife.rb.

**Comment by betamatt** [comment]
I've submitted a couple pull requests that are aimed at reducing dependence on internal conventions used in this project that would not be reliable if cluster management were distributed as an independent gem.

**Comment by mrflip** [comment]
+1 but I'd like to hold until we have the version_2 branch landed later this month

**Comment by mrflip** [comment]
I put the jeweler guts into the Rakefile and added a Gemfile.  Haven't actually shipped it as a gem (`rake release`), but I thought this would be helpful.

I made it exclude roles/, clusters/, cookbooks, etc from the gem -- if that's wrong, go to the Rakefile and edit

``` ruby
  gem.files = dotfiles + Dir["**/*"].
    reject{|f| f =~ %r{^(clusters|config|cookbooks|data_bags|roles|site-cookbooks)/} }. 
    reject{|f| File.directory?(f) }.
    #... 
```

**Comment by mrflip** [comment]
closed by pull request #59 and fiksu@719cbfe
